### PR TITLE
[Internal] Perf test: Refactors code to re-enable performance tests

### LINKED ISSUE #2840: [Internal] Performance Test: Fixs and optimize benchmark results
There is a fluctuation in numbers upto 25%. Tried below things:

Fix it by upgrading benchmarknotnet but it didn't work.
Different combinations of run strategies like medium with 2 iteration, medim wit 3 iteration, short with 2 iteration, short with 3 iteration, Custom launchcount, Warmupcount and Iteration Count
Assumption is, something is changed in the machine where these tests runs (because we didn't make any code changes recently in this project).

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

Performance tests were disabled due to inconsistency in results as part of this PR https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2838. Re-enabling it as part of this PR.

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

## Related Issue

Closes #2840

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json [hunk 1]
```diff
@@ -1,79 +1,79 @@
 ﻿{
-  "MasterKeyAuthorizationBenchmark.CreateSignatureGeneration;": 534.0,
-  "MasterKeyAuthorizationBenchmark.ReadSignatureGeneration;": 532.0,
-  "MockedItemBenchmark.CreateItem;[Type=OfT]": 45302.0,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json:4]: Higher is better right? 

Am seeing some of them already regression very heavily :-(

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json:4]: Can you please create a follow-up tasks for regressions for further investigation??

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json (hunk 1 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json:4]: no it is lower the better.. it is "allocated memory" numbers

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json [hunk 2]
```diff
@@ -1,79 +1,79 @@
...
-  "MockedItemBulkBenchmark.UpdateItem;[Type=OfTCustom]": 1195778.0,
-  "MockedItemBulkBenchmark.UpdateItem;[Type=OfTWithClientTelemetryEnabled]": 1235738.0,
-  "MockedItemBulkBenchmark.UpdateItem;[Type=OfTWithDiagnosticsToString]": 1231486.0,
-  "MockedItemBulkBenchmark.UpdateItem;[Type=Stream]": 773284.0,
-  "MockedItemBulkBenchmark.UpsertItem;[Type=OfT]": 1196210.0,
-  "MockedItemBulkBenchmark.UpsertItem;[Type=OfTCustom]": 1195590.0,
-  "MockedItemBulkBenchmark.UpsertItem;[Type=OfTWithClientTelemetryEnabled]": 1235596.0,
-  "MockedItemBulkBenchmark.UpsertItem;[Type=OfTWithDiagnosticsToString]": 1231380.0,
-  "MockedItemBulkBenchmark.UpsertItem;[Type=Stream]": 773312.0
+  "MasterKeyAuthorizationBenchmark.CreateSignatureGeneration;": "534",
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Performance.Tests/Contracts/BenchmarkResults.json:2]: Doesn't the change of data type (it was double now it is string) break any of the comparisons?

**Review by ealsur — APPROVED** [review]

**Pull request merged**